tests/apollo/test_skvbc_reconfiguration.py

```diff
@@ -187,6 +187,159 @@ async def test_semi_manual_upgrade(self, bft_network):
         await skvbc.assert_kv_write_executed(key, val)
         await skvbc.write_known_kv()
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    async def test_remove_nodes(self, bft_network):
+        """
+        In this test we show how a system operator can remove nodes (and thus reduce the cluster) from 7 nodes cluster
+        to 4 nodes cluster.
+        For that the operator performs the following steps:
+        1. Send a remove_node command - this command also wedges the system
+        2. Verify that all (including the removed candidates) have stopped
+        3. Load  a new configuration to the bft network
+        4. Rerun the cluster with only 4 nodes and make sure they succeed to perform transactions in fast path
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        client = bft_network.random_client()
+
+        key, val = await skvbc.write_known_kv()
+
+        await client.write(skvbc.write_req([], [], block_id=0, add_remove_node_command=True))
+
+        with trio.fail_after(seconds=90):
+            done = False
+            while done is False:
+                msg = skvbc.get_have_you_stopped_req(1)
+                rep = await client.read(msg, m_of_n_quorum=bft_client.MofNQuorum.All(client.config, [r for r in range(
```

Comment:
instead of `[r for r in range(...` can't you just say `bft_network.all_replicas()` ?

Reply:
done